src/remote/SockAddr.h

```diff
@@ -120,6 +122,44 @@ inline void SockAddr::checkAndFixFamily()
 	}
 }
 
+inline void SockAddr::convertFromMacOsToPosixWindows()
+{
+	struct
+	{
+		uint16_t sa_family;
+		char sa_data[14];
+	} sockAddrPosixWindows;
+
+	if (length() > sizeof(sockAddrPosixWindows))
+		fb_assert(false);
+	else
+	{
+		sockAddrPosixWindows.sa_family = ptr()->sa_family;
+		memcpy(sockAddrPosixWindows.sa_data, ptr()->sa_data, length() - 2);
+		memcpy(ptr(), &sockAddrPosixWindows, length());
+	}
+}
+
+inline void SockAddr::convertFromPosixWindowsToMacOs()
+{
+	struct
+	{
+		uint8_t sa_len;
+		uint8_t sa_family;
+		char sa_data[14];
+	} sockAddrMacOs;
+
+	if (length() > sizeof(sockAddrMacOs))
+		fb_assert(false);
+	else
+	{
+		sockAddrMacOs.sa_len = length();
+		sockAddrMacOs.sa_family = ptr()->sa_family;
+		memcpy(sockAddrMacOs.sa_data, ptr()->sa_data, length() - 2);
+		memcpy(ptr(), &sockAddrMacOs, length());
+	}
+}
+
```

Comment:
It will not work witn IPv6 addresses.

You don't need to copy sa_data at all ! 
Remove it from sockAddrPosixWindows\sockAddrMacOs and copy just 2 bytes. 
This is all you need.

Reply:
> It will not work witn IPv6 addresses.

Why?

Reply:
Because sizeof(sockaddr_in6) == 28 in both Windows and MacOS (i'm sure - on Linux too).
And you copy 22 bytes into 14.
And you don't need to copy more than first 2 bytes.